hypertrace-graphql-service/src/main/java/org/hypertrace/graphql/service/DefaultGraphQlServiceConfig.java

```diff
@@ -65,11 +73,27 @@
     this.entityServicePort = untypedConfig.getInt(ENTITY_SERVICE_PORT_PROPERTY);
     this.configServiceHost = untypedConfig.getString(CONFIG_SERVICE_HOST_PROPERTY);
     this.configServicePort = untypedConfig.getInt(CONFIG_SERVICE_PORT_PROPERTY);
-    // fallback timeout: 10s
+
     this.gatewayServiceTimeout =
-        untypedConfig.hasPath(GATEWAY_SERVICE_CLIENT_TIMEOUT)
-            ? untypedConfig.getDuration(GATEWAY_SERVICE_CLIENT_TIMEOUT)
-            : Duration.ofSeconds(10);
+        getSuppliedDurationOrFallback(
+            () -> untypedConfig.getDuration(GATEWAY_SERVICE_CLIENT_TIMEOUT));
+    this.attributeServiceTimeout =
+        getSuppliedDurationOrFallback(
+            () -> untypedConfig.getDuration(ATTRIBUTE_SERVICE_PORT_TIMEOUT));
+    this.configServiceTimeout =
+        getSuppliedDurationOrFallback(
+            () -> untypedConfig.getDuration(CONFIG_SERVICE_CLIENT_TIMEOUT));
+    this.entityServiceTimeout =
```

Comment:
Is this use anywhere?

Reply:
entityServiceTimeout is not being used anywhere. ```EntityTypeCachingClient``` uses a hardcoded value of 10s [here](https://github.com/hypertrace/entity-service/blob/88a7e4c0b77e538daf73663e092fbd3c6d067d40/entity-type-service-rx-client/src/main/java/org/hypertrace/entity/type/service/rxclient/EntityTypeCachingClient.java#L58). Should this value be replaced with this timeout?

Reply:
I was checking at the GQL level, if it is not used, should we add it or should we add it when it is used?

Reply:
Yes that's a fair point. I just added it so that we have a consistent configuration (as all other services had a timeout). Should I remove it (we do have a default fallback value though so even if you don't give the config it'll work).

Reply:
I think we can keep it for consistency as we are going with the different services that might have a different timeout requirement, and already have fields for (host, port) for this one.